src/ConstructiveSolidGeometry/VolumePrimitives/RegularPrism.jl

```diff
@@ -0,0 +1,90 @@
+struct RegularPrism{N,T,TR,TZ} <: AbstractVolumePrimitive{T}
+    r::TR
+    z::TZ
+    
+    function RegularPrism(  N::Integer,
+                            ::Type{T},
+                             r::Union{T, <:AbstractInterval{T}},
+                             z::Union{T, <:AbstractInterval{T}}) where {T}
+        @assert typeof(N) <: Integer && N >= 3 "Failed constructing RegularPrism with $(N) points in the polygon base (needs at least 3 points)."
+        new{N, T,typeof(r),typeof(z)}(r,z)
+    end
+end
+
+
+# Convenience functions
+const TriangularPrism{T,TR,TZ} = RegularPrism{3,T,TR,TZ}
+const QuadraticPrism{T,TR,TZ}  = RegularPrism{4,T,TR,TZ}
```

Comment:
I know this naming is consistent with the rest but I suggest we change it to SquarePrism to match the name more commonly used. I am calling `RegularPolygon{4,T}` a `Square{T}`